lib/HTFeed/DataDenZipAudit.pm

```diff
@@ -0,0 +1,169 @@
+#!/usr/bin/perl
+package HTFeed::DataDenZipAudit;
+
+use strict;
+use warnings;
+use HTFeed::DBTools qw(get_dbh);
+use HTFeed::Volume;
+use Carp;
+use Log::Log4perl;
+
+# Procedural function for randomly choosing a volume to check.
+#
+# my $vol = HTFeed::DataDenZipAudit->choose();
+# my $audit = HTFeed::DataDenZipAudit->new(namespace => $vol->{namespace},
+#                                          objid => $vol->{objid},
+#                                          path => $vol->{path},
+#                                          version => $vol->{version});
+# my $result = $audit->run();
+sub choose {
+  my $path_prefix = shift || '/htdataden';
+
+  my $sql = 'SELECT COUNT(*) FROM feed_backups' .
+            " WHERE path LIKE '$path_prefix%'" .
+            ' AND deleted IS NULL AND saved_md5sum IS NOT NULL';
+  my $dbh = HTFeed::DBTools->get_dbh();
+  my ($rows) = $dbh->selectrow_array($sql);
+  my $random_row = int(rand($rows));
+  my ($namespace, $objid, $path, $version);
+  $sql = 'SELECT namespace,id,path,version FROM feed_backups' .
```

Comment:
I believe we can do `order by rand() limit 1` -- the approach here might be faster, but I think probably not worth the increase in complexity in this case